src/main/java/seedu/address/logic/commands/CopyCommand.java

```diff
@@ -0,0 +1,178 @@
+        String personDetails = copyToClipboard(personToCopy, info);
+
+        return String.format(Messages.MESSAGE_COPIED, personDetails);
+    }
+
+    /**
+     * Copies the details of {@code personToCopy} to clipboard
+     */
+    private static String copyToClipboard(Person personToCopy, boolean[] info) {
+        requireNonNull(personToCopy);
+
+        StringBuilder copiedMsg = new StringBuilder();
+
+        if (info[0]) {
+            copiedMsg.append("Name: ").append(personToCopy.getName()).append("; ");
+        }
+        if (info[1]) {
+            copiedMsg.append("Phone: ").append(personToCopy.getPhone()).append("; ");
+        }
+        if (info[2]) {
+            copiedMsg.append("Email: ").append(personToCopy.getEmail()).append("; ");
+        }
+        if (info[3]) {
+            copiedMsg.append("Address: ").append(personToCopy.getAddress()).append("; ");
+        }
+        if (info[4]) {
+            copiedMsg.append("Tags: ").append(personToCopy.getTags()).append("; ");
+        }
+        if (info[5]) {
+            copiedMsg.append("Assets: ").append(personToCopy.getAssets()).append("; ");
+        }
+
+        return copiedMsg.substring(0, copiedMsg.length() - 2);
+    }
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the CopyCommand
+     * and returns an CopyCommand object for execution.
+     * @throws IllegalArgumentException if the user input does not conform the expected format
+     */
+    public static CopyCommand of(String args) throws IllegalArgumentException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL,
+                PREFIX_ADDRESS, PREFIX_TAG, PREFIX_ASSET);
+
+        Index index;
+
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (IllegalArgumentException ie) {
+            throw new IllegalArgumentException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, MESSAGE_USAGE), ie);
+        }
+
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS);
+
+        boolean[] info = new boolean[6];
+
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            info[0] = true;
+        }
+        if (argMultimap.getValue(PREFIX_PHONE).isPresent()) {
+            info[1] = true;
+        }
+        if (argMultimap.getValue(PREFIX_EMAIL).isPresent()) {
+            info[2] = true;
+        }
+        if (argMultimap.getValue(PREFIX_ADDRESS).isPresent()) {
+            info[3] = true;
+        }
+        if (argMultimap.getValue(PREFIX_TAG).isPresent()) {
+            info[4] = true;
+        }
+        if (argMultimap.getValue(PREFIX_ASSET).isPresent()) {
+            info[5] = true;
+        }
+
+        for (boolean b: info) {
+            if (b) {
+                return new CopyCommand(index, info);
+            }
+        }
```

Comment:
Copy command should only support one field at a time instead of allowing multiple fields.

Reply:
Copy command only supports 1 field now